source/MaterialXGenShader/TypeDesc.h

```diff
@@ -61,14 +62,21 @@ class MX_GENSHADER_API TypeDesc
 
     /// Empty constructor.
     constexpr TypeDesc() noexcept :
-        _id(0), _basetype(BASETYPE_NONE), _semantic(SEMANTIC_NONE), _size(0) { }
+        _id(0),
+        _basetype(BASETYPE_NONE),
+        _semantic(SEMANTIC_NONE),
+        _size(0),
+        _structIndex(0)
+    {
+    }
 
     /// Constructor.
-    constexpr TypeDesc(std::string_view name, uint8_t basetype, uint8_t semantic = SEMANTIC_NONE, uint16_t size = 1) noexcept :
+    constexpr TypeDesc(std::string_view name, uint8_t basetype, uint8_t semantic = SEMANTIC_NONE, uint8_t size = 1, uint8_t structIndex = 0) noexcept :
```

Comment:
Rather than focusing on efficient packing, perhaps it's better to make both of these values `uint_16_t` type, so that the hardcoded maximum number of struct indices is beyond what we would expect to see in any real-world asset.

I wouldn't expect to see any performance or memory consequences from instances of this class being one machine word larger!

Reply:
I'd defer to @niklasharrysson here - If I recall there was a comment/conversation with him where he was happy the size of this struct had not changed.  Perhaps related to the API changes introduced in MatX 1.39 where we're passing the `TypeDesc` object by value?

Reply:
Yeah, it was nice to fit the class into a single machine word, since these are passed by value. But I think you are right Jonathan, that on modern hardware, and compared to everything else going on in the API, there should be no noticeable difference to make it slightly larger.

So it's probably better to be on the safe side and make sure `size` and `structIndex` will never overflow.